src/main/scanLogic/graphScanLogic.ts

```diff
@@ -21,68 +22,115 @@ import { AbstractScanLogic } from './abstractScanLogic';
 export class GraphScanLogic extends AbstractScanLogic {
     public async scanAndCache(
         progress: vscode.Progress<{ message?: string; increment?: number }>,
-        componentsToScan: Set<ComponentDetails>,
+        Components: Components[],
         checkCanceled: () => void
     ) {
-        let graphResponse: IGraphResponse = await this._connectionManager.scanGraph(
-            componentsToScan,
-            progress,
-            checkCanceled,
-            Configuration.getProjectKey()
-        );
-        let scannedComponents: Map<string, INodeInfo> = new Map();
-        let licenses: Dictionary<string, ILicenseCacheObject> = new Dictionary();
-        let issues: IIssueCacheObject[] = [];
+        for (const componentsToScan of Components) {
+            let graphResponse: IGraphResponse = await this._connectionManager.scanGraph(
```

Comment:
IMO this is a bad idea. We should reduce the number of requests to Xray. I expect a major performance issue here.